citre-lang-c.el

```diff
@@ -98,6 +98,14 @@
           (setq syntax (if (eq syntax 'member)
                            'callable-member
                          'function))))
+      (save-excursion
+        (goto-char (car bounds))
+        (when (progn (backward-sexp 1) (not (eq (point) (car bounds))))
```

Comment:
Can we avoid using `backward-sexp`? I'm dealing with it a lot recently, some `forward-sexp-function`s will not take you to the previous symbol, but goes further to the opening delimiter of some bigger sexp (like in `web-mode`).

Maybe it works well in `c-mode`, but I consider it to be generally unreliable.

I think you could do one of:

- `looking-back` a regexp, like `(rx (group (seq symbol-start (* anything) symbol-end)) (* (or space "\n")))` I didn't test it though.
- Move backward blanks and then grab the symbol. You can use `puni--backward-blanks` in https://github.com/AmaiKinono/Tokimacs/blob/master/site-lisp/puni.el.

Reply:
Updated. I used looking-back.